src/login/LoginForm.js

```diff
@@ -48,47 +55,48 @@ function LoginForm() {
 
   return (
     <>
-      <form className={classes.root}>
-        <TextField
-          data-testid="test123"
-          label="Username"
-          name="username"
-          id="username"
-          value={username}
-          onChange={(e) => setUsername(e.target.value)}
-        />
-        <br />
-        <TextField
-          label="Password"
-          name="password"
-          id="password"
-          type="password"
-          value={password}
-          onChange={(e) => setPassword(e.target.value)}
-        />
-        <br />
-        <Button variant="contained" color="primary" disabled={!valid} onClick={loginUser}>
-          Login
-        </Button>
-      </form>
-      <Backdrop open={loading} className={classes.backdrop}>
-        <CircularProgress color="primary" />
-      </Backdrop>
-      <Snackbar open={called && !!user} data-testid="login--success-alert">
-        <Alert severity="success">
-          You&apos;re logged in!
-        </Alert>
-      </Snackbar>
-      <Snackbar open={called && !user && !error} data-testid="login--info-alert">
-        <Alert severity="info">
-          You don&apos;t have an account!
-        </Alert>
-      </Snackbar>
-      <Snackbar open={called && !!error} data-testid="login--error-alert">
-        <Alert severity="error">
-          An error occured!
-        </Alert>
-      </Snackbar>
+      {state.user
```

Comment:
This approach to checking authentication is really hard to scale and maintain. Instead check authentication when defining the route, something like https://www.bradcypert.com/auth-guarding-react-router-routes/